### ISSUE TITLE
Some Android features launch Chrome instead of the user’s default browser

### ISSUE THREAD

**Issue by csadilek** [mozilla/platform-tilt#8]
Features like Google Search, or Discover, in the pre-installed [Google application](https://play.google.com/store/apps/details?id=com.google.android.googlequicksearchbox) ignore the user’s default browser choice: links to websites outside of the application are always opened in Chrome, regardless of the default browser. This is a widely used application, with additional entry points from built-in features such as the search bar on the home screen and app launcher. Each time it opens a link in Chrome, a user is driven away from their default browser.

All built-in applications and affordances that open external links should open them in the user’s default browser.

**Comment by S0ulf3re** [comment]
In Google Search within the Google app, the default browser can be set to open links in the browser by going to profile icon -> settings -> general, then toggling "Open web pages in the app" off. This will instead send them to your browser app.

**Comment by lol768** [comment]
> In Google Search within the Google app, the default browser can be set to open links in the browser by going to profile icon -> settings -> general, then toggling "Open web pages in the app" off. This will instead send them to your browser app.

The UX is strictly worse if you do this. Firefox on Android has perfectly good support for the Android Custom Tabs protocol which works just fine with most other apps and allows for webpages to be opened in a lightweight view that inherits my settings, add-ons etc and that I can launch into a full Firefox Android session _if I want to._ If you follow these instructions, it will use a basic intent and launch the app separately which will pull you out of the flow of the search results or content feed (for e.g. Discover).

Every other app apart from the crap Google ship is capable of using `resolveActivity` to figure out the user's default browser and then using _that_ for custom tabs support if the user's default browser app supports it. It's about [3 lines of code](https://github.com/UniversityofWarwick/postroom-android/blob/efb865359e0238bf1b1b7efd31523c5b1acb9ad0/app/src/main/java/uk/ac/warwick/postroom/services/CustomTabsServiceImpl.kt#L50-L54); there's no excuse.

**Comment by UndarkAido** [comment]
AFAIK it's impossible to use Firefox in Google News and Meta apps

**Comment by LinAGKar** [comment]
Don't know if it fits here, but Steam on Android also forces users over to Chrome instead of the default browser when opening links.

**Comment by UndarkAido** [comment]
@LinAGKar Some of Steam's links open in the default browser and some force Chrome, weird

**Comment by resuna** [comment]
I am regularly copying and pasting URLs from Chrome to Firefox because of this.

**Comment by ggould256** [comment]
Fascinatingly if you uninstall Chrome from the device then Discover, at least, _does_ respect the default browser -- that is, it tries Chrome and if that isn't available it falls back to the default browser.
So it's not even that they've missed some crucial call to respect the default browser.

**Comment by mstarongithub** [comment]
> Fascinatingly if you uninstall Chrome from the device then Discover, at least, _does_ respect the default browser -- that is, it tries Chrome and if that isn't available it falls back to the default browser. So it's not even that they've missed some crucial call to respect the default browser.

Yep. Other apps suddenly respect your default browser too after removing Chrome (be it disable or uninstall), even things like the article view on the Pixel series' launcher

**Comment by UndarkAido** [comment]
@ggould256 Tap your avatar > settings > other settings > disable "Open web pages in the app"

**Comment by ading2210** [comment]
Many Android apps made by Google outright refuse to open links in browsers other than Chrome. The Google Classroom app is an example of this: 

![image](https://github.com/mozilla/platform-tilt/assets/71154407/b014c8bb-2f53-4999-85f8-2a49a52d0d52)

**Comment by sibbl** [comment]
Google News is another example of this. Even when disabling the Chrome app in Android, it shows websites in some embedded version of Chrome instead of using the default browser as I'd expect it to do. There's no setting to disable this behavior.

**Comment by rstat1** [comment]
> AFAIK it's impossible to use Firefox in Google News and Meta apps

For Google News clicking the View Original Webpage link in the menu of an article opens it whatever your default browser is. So it's not completely impossible

**Comment by lol768** [comment]
> For Google News clicking the View Original Webpage link in the menu of an article opens it whatever your default browser is. So it's not completely impossible

Right, which you have to do once it's already opened and rendered in-app using Chrome, for each and every article. It also launches a standard intent instead of a lightweight custom tab hosted within Google News

**Comment by youngacinonyx** [comment]
Hi! I was having trouble creating a widget to open Google News using Firefox, but I managed to find a workaround. While it's kinda ugly and not the ideal solution, it works, so I thought it was worth sharing. Here's what I did:

First, go to the settings of your phone and make sure Firefox has permission to add icons to the home screen. In my case, I had to follow this path: _Settings - Apps - Manage apps - Firefox - Other permissions - Home screen shortcuts - Always allow_

After that, follow these steps:
- Disconnect from any network (wifi and mobile data)
- Open Firefox
- Open a new (empty) tab
- Type _news.google.com_ in the address bar and confirm
- Tap on the three-dotted icon on the right of the address bar
- Tap on _Add to Home screen_
- Change the name to _News_ or something similar
- Tap on _Add_
- Open Firefox again
- Close the tab you've previously opened

If everything worked correctly, you should have a new icon that opens Google News inside Firefox. For some reason, if you try to do this while connected to a network, you will be able to create the icon, but it will not open inside Firefox (even if the native Google News app is uninstalled and both Chrome and Google apps are disabled).

**Comment by Dudes1** [comment]
Google app, settings, other settings. I turned off "Open web pages in the app" and it now opens links in my default browser (not Chrome). Pixel 8a Android 15

**Comment by ralphm** [comment]
> Google app, settings, other settings. I turned off "Open web pages in the app" and it now opens links in my default browser (not Chrome). Pixel 8a Android 15

The Google app has had this for a while. The Google *News* app doesn't.

**Comment by Dudes1** [comment]
Oops sorry
